modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/model/listener/UserGroupModelListener.java

```diff
@@ -58,14 +60,22 @@ private void _unsubscribeUserFromUserGroupGroups(
 			long userId, long userGroupId)
 		throws PortalException {
 
-		List<Group> userGroupGroups = _groupLocalService.getUserGroupGroups(
+		long[] userGroupGroupIds = _userGroupLocalService.getGroupPrimaryKeys(
 			userGroupId);
 
-		userGroupGroups.removeAll(
-			_groupLocalService.getUserGroups(userId, true));
+		List<Group> userGroupsList = _groupLocalService.getUserGroups(
+			userId, true);
 
-		for (Group group : userGroupGroups) {
-			_blogsEntryLocalService.unsubscribe(userId, group.getGroupId());
+		Set<Long> userGroups = new HashSet<>(userGroupsList.size());
+
+		for (Group group : userGroupsList) {
+			userGroups.add(group.getGroupId());
+		}
+
+		for (long groupId : userGroupGroupIds) {
+			if (!userGroups.contains(groupId)) {
+				_blogsEntryLocalService.unsubscribe(userId, groupId);
+			}
 		}
 	}
 
```

Comment:
@marianoalvarosaiz, it looks like `Set<Long> userGroups ` is unnecessary. We may simplify this 

```
long[] userGroupGroupIds = _userGroupLocalService.getGroupPrimaryKeys(
			userGroupId);

List<Group> userGroups = _groupLocalService.getUserGroups(userId, true);

for (long groupId : userGroupGroupIds) {
		if (!userGroups.contains(groupId)) {
				_blogsEntryLocalService.unsubscribe(userId, groupId);
			}
		}
```